python/ccf/ledger.py

```diff
@@ -671,7 +672,18 @@ def __init__(self, filename: str):
             receipt_bytes = _peek_all(self._file, pos=receipt_pos)
 
             receipt = json.loads(receipt_bytes.decode("utf-8"))
-            root = ccf.receipt.root(receipt["leaf"], receipt["proof"])
+            if "leaf" in receipt:
+                leaf = receipt["leaf"]
```

Comment:
I'm assuming that this path is only used since we've added receipts in snapshots (#2998) but hasn't been included in an official (i.e. non-dev) release just yet. Is my understanding correct?

Reply:
Yes, this is for compatibility with pre-claims snapshots, and so I think we should remove it from 2.0 final. What do you think?

Reply:
Removing it sounds like a good idea!